### ISSUE TITLE
bigo.tv plugin - cannot seem to play saved file

### ISSUE THREAD

**Issue by bigotvuser** [streamlink/streamlink#1031]
- [x] This is a bug report.
- [ ] This is a feature request.
- [ ] This is a plugin (improvement) request.
- [ ] I have read the contribution guidelines.

### Description
Hey  guys, the people at bigo.tv have made changes again and now the streams are not playing and saved files open but do not show any content. Could require a similar fix as in issue 234. . 
...

### Expected / Actual behavior
..

### Reproduction steps / Explicit stream URLs to test

streamlink http://bigoweb.co/show/2210868181 best -o kabbi1nes.flv
I tried the latest streamlink version 0.6.0 and see the same results. 
[kabbi1nes.zip](https://github.com/streamlink/streamlink/files/1098918/kabbi1nes.zip)

### Environment details

Operating system and version: ...Win 10
Streamlink and Python version: ..streamlink - .0.6.0

...

### Comments, logs, screenshots, etc.

...

**Comment by unnutricious** [comment]
I can confirm that I see the same thing on any Bigo stream. I haven't figured out what has changed in the protocol yet, though (I wasn't very familiar with the original one). It still seems to be streaming data with FLV headers, but it's not valid.

The flash parameters all look the same as before, but I am guessing we need to send a different set of parameters after opening the socket now.

**Comment by unnutricious** [comment]
I confirmed that the 'login' protocol has not changed at all. The server is just now sending back different data. There are an extra 16 bytes pre-pending before the FLV stream starts, which causes the player to not recognize it. If you strip these bytes out, you can play the first frames of the video. However, there still appear to be errors in the stream, so I suspect it is inserting extra non-FLV data elsewhere as well. I am not that familiar with FLV, though, so it's hard to tell.

After stripping the first 16 bytes from the example capture, flvmeta reports these errors:
```
0x0000014b: warning W70044: 3 bytes missing from tag body after metadata end
0x0000014f: error E12051: previous tag size should be 317, 81160 found instead
0x0000014f: error E30013: unknown tag type 0
0x00000157: error E20025: tag stream id must be zero, 175 found instead
0x0000055e: error E12051: previous tag size should be 1035, 1694485086 found instead
0x0000055e: error E30013: unknown tag type 212
0x0000055e: fatal F30084: consecutive tags with unknown type found, aborting
6 error(s), 1 warning(s)
```

**Comment by back-to** [comment]
the easy way would be to remove the flash stream and use **HLS** instead

with the **IPhone** User-Agent you will get something like

```html
        <div class="video_room_box" id="video_room_box_e">
            <div class="video_content" id="room_video_e" video-type="1" chat-value="http://8.8.8.8:7777/show123&123&123?asdf">
                <div class="video_landscape" id="video_landscape_e">
                    <video class="live-video vjs-default-skin" id="video_tag_e" data-setup="{}" webkit-playsinline playsinline controls="controls" preload="none" width="100%" height="100%" poster="">
                        <source id="videoSrc" src="http://8.8.8.8:7777/list_8888888_8888888.m3u8" type="application/x-mpegURL">
                    </video>
                </div>
            </div>
```

---

#843

**Comment by unnutricious** [comment]
@back-to thanks for the idea, that worked perfectly. I just added HLS support in hopes that flv streaming can still be repaired.

**Comment by bigotvuser** [comment]
this is great stuff guys. thanks. Hoping the build is successful so that I can give it a go.

**Comment by back-to** [comment]
originale solved with #1053
